listener.go

```diff
@@ -263,12 +263,18 @@ func (l *listener) Accept() (net.Conn, error) {
 	l.init.Do(func() {
 		if len(l.sans) > 0 {
 			if err := l.updateCert(l.sans...); err != nil {
-				logrus.Errorf("failed to update cert with configured SANs: %v", err)
+				logrus.Errorf("dynamiclistener %s: failed to update cert with configured SANs: %v", l.Addr(), err)
```

Comment:
```suggestion
				logrus.Errorf("dynamiclistener %s: failed to update cert with configured SANs: %v", l.Addr().String(), err)
```

Reply:
These implement the Stringer interface so explicitly calling `String()` alongside a format string is unnecessary.
https://go.dev/tour/methods/17

Reply:
Glad to change it if there's a code style guideline I'm missing. There are a couple other instances of this as well here that I can change if necessary.

Reply:
Nope, I missed that mb.